Tasks/AzureFunctionV2/src/request-builder.ts

```diff
@@ -0,0 +1,95 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import { TaskInputs } from './types';
+import { AxiosRequestConfig } from 'axios';
+import { AzureRMEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azure-arm-endpoint';
+import { AuthType } from './constants';
+import { AzureEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azureModels';
+
+export class RequestBuilder {
+  constructor(private inputs: TaskInputs) {}
+
+  public async buildRequest(): Promise<AxiosRequestConfig> {
+    return this.inputs.authType === AuthType.Key ? this.buildKeyAuthRequest() : await this.buildArmAuthRequest();
+  }
+
+  private buildKeyAuthRequest(): AxiosRequestConfig {
+    tl.debug(`Building key authentication request for function: ${this.inputs.function}`);
+    let url = `${this.inputs.function}?code=${this.inputs.key}`;
+
+    // Add query parameters if present
+    if (this.inputs.queryParameters) {
+      url += this.inputs.queryParameters;
+    }
+
+    // Merge system headers with user-provided headers
+    const systemHeaders = this.getDefaultHeaders();
+    const headers = {
+      ...systemHeaders,
+      ...this.inputs.headers // User headers override system headers
+    };
+
+    const config: AxiosRequestConfig = {
+      url: url,
+      method: this.inputs.method,
+      headers,
+      validateStatus: () => true, // Don't throw for any HTTP status, we'll handle it manually
+      data: this.getRequestBody()
+    };
+
+    return config;
+  }
+
+  private async buildArmAuthRequest(): Promise<AxiosRequestConfig> {
+    if (!this.inputs.serviceConnection) {
+      throw new Error('Service connection is required for ARM authentication');
+    }
+
+    tl.debug(`Building request with ARM authentication using service connection: ${this.inputs.serviceConnection}`);
+
+    try {
+      const request = this.buildKeyAuthRequest();
```

Comment:
This means that request url will contain `code` query parameter even in cases when task is using ARM service connection and key is empty. I suggest to extract the common request builder logic responsible for both ARM auth and key auth into separate method and call in both `buildArmAuthRequest` and `buildKeyAuthRequest`.

Reply:
Azure functions always require a key, so no matter if we use `Key Auth` or `ARM Auth`, we still need to pass the key.
That's why Key auth should be used as base auth way and ARM is built on top of it.